File: Legacy/data/figure/plot.R
library("ggpubr")

dat_inv <- read.csv("../inv.csv")
dat_det <- read.csv("../det.csv")
dat_add <- read.csv("../add.csv")
dat_mul <- read.csv("../mul.csv")

dat <- rbind(dat_add, dat_mul, dat_det, dat_inv)

lib <- c("Peroxide(Rust)", "DNumeric(D)", "R", "Numpy(Python)", "Julia")
Library <- rep(lib, 4)

work <- c(rep("add", 5), rep("mul", 5), rep("det", 5), rep("inv", 5))

df <- cbind(dat, Library, work)

g <- ggline(df, "work", "mean", type='l', color="Library", ylab="time(s)", title="Benchmark for matrix operations")
g + yscale("log10", .format=TRUE)

File: total/result/plot.py
import pylab as plt
import pandas as pd

add = pd.read_csv("add.csv")
mul = pd.read_csv("mul.csv")
det = pd.read_csv("det.csv")
inv = pd.read_csv("inv.csv")
sol = pd.read_csv("solve.csv")

rust = [add["mean"][0], mul["mean"][0], det["mean"][0], inv["mean"][0], sol["mean"][0]]
julia = [add["mean"][1], mul["mean"][1], det["mean"][1], inv["mean"][1], sol["mean"][1]]
python = [add["mean"][2], mul["mean"][2], det["mean"][2], inv["mean"][2], sol["mean"][2]]

# Use latex
plt.rc('text', usetex=True)
plt.rc('font', family='serif')

# Plot
plt.figure(figsize=(10,6), dpi=300)
plt.title(r"Benchmark of Matrix operations", fontsize=16)
plt.xlabel(r'Operations', fontsize=14)
plt.ylabel(r'time(sec)', fontsize=14)


temp = [1, 2, 3, 4, 5]
plt.xticks(temp, ('Add', 'Mul', 'Det', 'Inv', 'Solve'))
plt.plot(temp, rust, label='Peroxide')
plt.plot(temp, julia, label='Julia')
plt.plot(temp, python, label='Numpy')

plt.legend(fontsize=12)
plt.grid()
plt.savefig("matrix_result.png", dpi=300)
